Dockerfile

```diff
@@ -0,0 +1,112 @@
+        json-c-dev \
+        libmilter-dev \
+        libtool \
+        libxml2-dev \
+        linux-headers \
+        make \
+        ncurses-dev \
+        openssl-dev \
+        pcre2-dev \
+        python3 \
+        zlib-dev \
+    && \
+    mkdir -p "./build" && \
+    cd "./build" && \
+    cmake .. \
+          -DAPP_CONFIG_DIRECTORY="/etc/clamav" \
+          -DCMAKE_BUILD_TYPE"=release" \
+          -DCMAKE_INSTALL_FULL_BINDIR="/usr/bin" \
+          -DCMAKE_INSTALL_FULL_INCLUDEDIR="/usr/include" \
+          -DCMAKE_INSTALL_FULL_LIBDIR="/usr/lib" \
+          -DCMAKE_INSTALL_FULL_SBINDIR="/usr/sbin" \
+          -DCMAKE_INSTALL_LIBDIR="/usr/lib" \
+          -DCMAKE_INSTALL_PREFIX="/usr" \
+          -DENABLE_CLAMONACC=OFF \
+          -DENABLE_DOCS=OFF \
+          -DENABLE_EXAMPLES=OFF \
+          -DENABLE_JSON_SHARED=ON \
+          -DENABLE_MILTER=ON \
+          -DENABLE_STATIC_LIB=OFF && \
+    make DESTDIR="/clamav" -j$(($(nproc) - 1)) install && \
+    rm -r \
+       "/clamav/usr/include" \
+       "/clamav/usr/lib/pkgconfig/" \
+    && \
+    sed -e "s|^\(Example\)|\# \1|" \
+        -e "s|.*\(PidFile\) .*|\1 /run/lock/freshclam.pid|" \
+        -e "s|.*\(DatabaseOwner\) .*|\1 clamav|" \
+        -e "s|^\#\(UpdateLogFile\) .*|\1 /var/log/clamav/freshclam.log|" \
+        -e "s|^\#\(NotifyClamd\).*|\1 /etc/clamav/clamd.conf|" \
+        -e "s|^\#\(ScriptedUpdates\).*|\1 yes|" \
+        -e "s|^\#\(AllowSupplementaryGroups\).*|\1 yes|" \
+        "/clamav/etc/clamav/freshclam.conf.sample" > "/clamav/etc/clamav/freshclam.conf" && \
+    sed -e "s|^\(Example\)|\# \1|" \
+        -e "s|.*\(PidFile\) .*|\1 /run/lock/clamd.pid|" \
+        -e "s|.*\(LocalSocket\) .*|\1 /run/clamav/clamd.sock|" \
+        -e "s|.*\(TCPSocket\) .*|\1 3310|" \
+        -e "s|.*\(TCPAddr\) .*|\1 0.0.0.0|" \
+        -e "s|.*\(User\) .*|\1 clamav|" \
+        -e "s|^\#\(LogFile\) .*|\1 /var/log/clamav/clamd.log|" \
+        -e "s|^\#\(LogTime\).*|\1 yes|" \
+        -e "s|^\#\(AllowSupplementaryGroups\).*|\1 yes|" \
+        "/clamav/etc/clamav/clamd.conf.sample" > "/clamav/etc/clamav/clamd.conf" && \
+    sed -e "s|^\(Example\)|\# \1|" \
+        -e "s|.*\(PidFile\) .*|\1 /run/lock/clamav-milter.pid|" \
+        -e "s|.*\(MilterSocket\) .*|\1 inet:7357|" \
+        -e "s|.*\(User\) .*|\1 clamav|" \
+        -e "s|.*\(\ClamdSocket\) .*|\1 unix:/run/clamav/clamd.sock|" \
+        "/clamav/etc/clamav/clamav-milter.conf.sample" > "/clamav/etc/clamav/clamav-milter.conf"
+
+FROM registry.hub.docker.com/library/alpine:latest
+
+RUN apk add --no-cache \
+        fts \
+        json-c \
+        libbz2 \
+        libcurl \
+        libltdl \
+        libmilter \
+        libstdc++ \
+        libxml2 \
+        ncurses-libs \
+        pcre2 \
+        tini \
+        zlib \
+    && \
+    addgroup -S clamav && \
+    adduser -D -G "clamav" -h "/var/lib/clamav" -s "/bin/false" -S "clamav" && \
+    install -d -m 755 -g "clamav" -o "clamav" "/var/log/clamav"
+
```

Comment:
Please add a call to run `freshclam` at the end here so the image contains the databases and users don't have to download the CVDs when they start.  Running `freshclam` in the `init` script is still important though, as it will update the CVD's using the CDIFF patching process, which uses far-far less data.

Reply:
as discussed above, will do!

Reply:
I don't think FreshClam fails (non-zero exit) when `NotifyClamd` fails (eg because ClamD isn't running).   I'm pretty sure it merely prints a warning.  I don't think you need to worry about it.

Reply:
it is only a warning; but if there's a flag to silence it, it would be nice :) As I looked into the code, there IS a NULL check to avoid the warning, but I can't seem to trigger it. Maybe a --no-daemon-notify option to work around that in the (far) future :p

Reply:
The easy solution would be to only add `NotifyClamd` to the `freshclam.conf` after the first time FreshClam is run.  I don't think a new CLI option would be super helpful here.

Reply:
ahh, it's set in freshclam.conf! So it is indeed, not null. Can we run freshclam without a config file? because in that case, NotifyClamd would be null, right? If we can resolve this all with command line parameters we're good too. E.g. path being the crucial bit.

You are right, this is not the most important thing, but if we can get rid of the warning in a 'nice and clean' way ... :)

Alternatively, expand the error handling to check if the filepath of clamd != '/dev/null' :)

I know I should just dig into it and figure out what works, but I'm a bit lazy and asking the expert here ;)

Reply:
You could comment out `NotifyClamd` in the config file or remove it and then add it after the first run.  Or you could use the `freshclam --config-file=FILE` option to user a different config.  In a future version (maybe 0.104, _hopefully_ 0.104) we'd like to make it so a config file is not required, but for now you at least need a config file that contains the `DatabaseMirror` setting.   

Personally I'd recommend just ignoring it.  Our image users will never see it -- only people building the image will notice it.  In the case of us building the image, that'll end up being automated anyways, so we won't _really_ notice it.